plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,180 @@
+		$info                  = array(
+			'error_message' => $result['description'],
+			'error_code'    => $response->get_error_code(),
+			'available'     => false,
+		);
+	} else {
+		$status_code = wp_remote_retrieve_response_code( $response );
+		$data        = json_decode( wp_remote_retrieve_body( $response ), true );
+		$info        = array(
+			'status_code'   => $status_code,
+			'available'     => false,
+			'error_message' => '',
+		);
+
+		if (
+			400 === $status_code &&
+			isset( $data['data']['params'] ) &&
+			is_array( $data['data']['params'] ) &&
+			count( $data['data']['params'] ) > 0
+		) {
+			// The REST API endpoint is available.
+			$info['available'] = true;
+		} elseif ( 401 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Authorization should not be required to access the Optimization Detective REST API endpoint.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'To collect URL metrics, the Optimization Detective REST API endpoint should be accessible without requiring authorization.', 'optimization-detective' );
+		} elseif ( 403 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Access to the Optimization Detective REST API endpoint was denied.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'Access was denied because the user does not have the necessary capabilities. Please check user roles and capabilities, as all users should have access to the Optimization Detective REST API endpoint.', 'optimization-detective' );
+		} else {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Error accessing the Optimization Detective REST API endpoint', 'optimization-detective' );
+			$result['description'] = esc_html__( 'There was an issue reaching the Optimization Detective REST API endpoint. This might be due to server settings or the REST API being disabled.', 'optimization-detective' );
+		}
+		$info['error_message'] = $result['description'];
+	}
+
+	$result['description'] = sprintf(
+		'<p>%s</p>',
+		$result['description']
+	);
+
+	update_option( 'od_rest_api_info', $info );
+	return $result;
+}
+
+/**
+ * Renders an admin notice if the REST API health check fails.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<string> $additional_classes Additional classes to add to the notice.
+ */
+function od_render_rest_api_health_check_notice( array $additional_classes = array() ): void {
+	$rest_api_info = get_option( 'od_rest_api_info', array() );
+	if (
+		isset( $rest_api_info['available'] ) &&
+		false === $rest_api_info['available'] &&
+		isset( $rest_api_info['error_message'] ) &&
+		'' !== $rest_api_info['error_message']
+	) {
+		wp_admin_notice(
+			esc_html( $rest_api_info['error_message'] ),
```

Comment:
TODO: We need to append a message here to say something like, "Please visite Site Health to re-check once you have addressed the issue."

Reply:
Done in 4c1b2dc35e1d64951f2bb33b35cda7ee971a8cab